typeclasses/vehicles.py

```diff
@@ -0,0 +1,30 @@
+"""
+Vehicle Objects are specifically used for objects 
+where a character can enter it, exit it, and (possibly)
+have some method to drive it, if it's not automatic.
+
+Door code for vehicles is also contained, where they
+possess verbs to exit the object to the parent.
+"""
+
+from evennia import DefaultObject
+from commands.vehicles import CmdSetVehicle
+
+class VehicleObject(DefaultObject):
+
+    def at_object_creation(self):
+        # Vehicles will need a script to enter them
+        # Without having to use @tel, like we have been
+        # And having a door object in the area, seperate
+        # Of the ship/loincloth is ineffective.
+
+        self.cmdset.add_default(CmdSetVehicle)
+
+
+class VehicleDoorObject(DefaultObject):
+
+    def at_object_creation(self):
+        # Ideally, I want to have this object in a state
+        # Where it can be set down, and be ready for use.
+
+        self.cmdset.add_default(CmdSetVehicle)
```

Comment:
shouldn't this just be given to characters? (although I'm not familiar with how cmdsets work tbh)

Reply:
okok so cool teaching experience here, so basically in the server/conf/lockfuncs, we can set barriers to how commands can and can't work. that code is:

def cmdLocationCheck(accessing_obj, accessed_obj, *args, **kwargs):
    """
    Usage: cmdLocationCheck()
    A simple check to see if the player(accessing_obj) is in the
    same location/room as the acccessed_obj, which could be a
    vehicle, door, or anything inbetween.
    """
    accessorParent = accessing_obj.location
    accessedParent = accessed_obj.location
    return accessedParent == accessorParent
    
We have the accessor, the player accessing the command, and the accessed, which is the object. That accessed object is decided by where the command set is put in, so the door and vehicle. Right now technically you could "enter the door" and be trapped in a door dimension, but i gotta figure out how to maybe only use 1 command from a cmdset before i separate them.